src/main/scala/it/pagopa/interop/catalogprocess/common/readmodel/ReadModelQueries.scala

```diff
@@ -96,6 +98,26 @@ object ReadModelQueries {
 
   def emptyResults[T] = PaginatedResult[T](results = Nil, totalCount = 0)
 
+  def getEServiceDocument(eServiceId: UUID, descriptorId: UUID, documentId: UUID)(
+    readModel: ReadModelService
+  )(implicit ec: ExecutionContext): Future[Option[CatalogDocument]] = {
+    val filters = Filters.and(
+      Filters.eq("data.id", eServiceId.toString),
+      Filters.eq("data.descriptors.id", descriptorId.toString),
+      Filters.or(
+        Filters.eq("data.descriptors.docs.id", documentId.toString),
+        Filters.eq("data.descriptors.interface.id", documentId.toString)
+      )
+    )
+    readModel.findOne[CatalogDocument]("eservices", filters)
```

Comment:
uhm are you sure this is working? the query should return a `CatalogItem` but it is trying to decode it as `CatalogDocument`

Reply:
I need to implement a aggregateRaw sorry.

Reply:
Better to do a filtering on CatalogItem